src/templates/index.jsx

```diff
@@ -299,7 +300,7 @@ const Layout = React.createClass({
 
         /* If we are in detail view, move to next note */
         if (this.props.selectedNoteId) {
-            return this.props.selectNote(filteredNotes[newIndex].id);
+            return this.props.selectNote(filteredNotes[newIndex].id, this.props.filterName);
```

Comment:
Could just be my misunderstanding of redux :) Should I just access the filterName directly in the  reducer?